src/frequenz/lib/notebooks/reporting/utils/helpers.py

```
@@ -0,0 +1,88 @@
+# License: MIT
+# Copyright © 2025 Frequenz Energy-as-a-Service GmbH
+"""Helper functions for Microgrid Data Processing Utilities.
+
+This module provides utility functions for preprocessing and analyzing microgrid
+data represented in pandas DataFrames. It derives photovoltaic (PV) energy flows
+from the given dataframe.
+
+Key Features
+------------
+- PV Energy Flow Calculations
+  Derives PV production, excess, self-consumption, battery charging, and
+  grid feed-in metrics, including PV self-consumption share.
+
+Usage
+-----
+These functions serve as building blocks for energy reporting, data pipelines,
+and dashboards that analyze microgrid performance, particularly in hybrid systems
+with PV, batteries, and grid interactions.
+"""
+
+import pandas as pd
+
+
+def _add_pv_energy_flows(df: pd.DataFrame) -> pd.DataFrame:
+    """Add PV-related energy flow columns to ``df`` if PV data is present.
+
+    Derives photovoltaic (PV) energy-flow metrics from existing columns. If no PV
+    signal is present (i.e., the negative PV column is missing or all zeros), the
+    DataFrame is returned unchanged.
+
+    Args:
+      df: Input DataFrame. If present, uses columns ``pv_neg``,
+        ``consumption``, and ``battery_pos``. Missing columns are
+        treated as zeros.
```

I would raise if any of the expected columns is missing, otherwise it's easy to create typo bugs.

---

Yeah, I picked it from the notebooks. I think previously sometimes consumption column would be missing so this logic was implemented but now we should raise if any column name is missing.
But one question would be whether pv_columns and battery columns will be present in the data even if the component is missing?

---

So, I think I need to add the logic to check whether the component type is present in the data. Then forward with the checks/calculations otherwise pass.